src/OrbitGl/App.cpp

```diff
@@ -2890,19 +2890,25 @@ void OrbitApp::SelectCallstackEvents(const std::vector<CallstackEvent>& selected
 void OrbitApp::InspectCallstackEvents(const std::vector<CallstackEvent>& selected_callstack_events,
                                       bool origin_is_multiple_threads) {
   SetCaptureDataSelectionFields(selected_callstack_events, origin_is_multiple_threads);
-  main_window_->SetCallTreeInspection(CallTreeView::CreateTopDownViewFromPostProcessedSamplingData(
-                                          GetCaptureData().selection_post_processed_sampling_data(),
-                                          *module_manager_, GetCaptureData()),
-                                      CallTreeView::CreateBottomUpViewFromPostProcessedSamplingData(
-                                          GetCaptureData().selection_post_processed_sampling_data(),
-                                          *module_manager_, GetCaptureData()));
+  auto report = std::make_shared<SamplingReport>(
```

Comment:
can this be a unique pointer that we move into the main_window_?

Reply:
Done; in the the end, it does get turned back into a shared_ptr because for general selection, it is shared between App and orbitsamplingreport, but since we're not doing that here I kept it as a unique_ptr until it had to move into a shared_ptr.

Reply:
I see. We might want to reconsider also having a clear ownership between App and orbitsamplingreport, but thats a different story. I still think here it makes sense to have it as a unique_ptr.